python/API_GW_REQUIRE_AUTHORIZER/API_GW_REQUIRE_AUTHORIZER.py

```diff
@@ -0,0 +1,505 @@
+        api_id = configuration_item.get("resourceId").split("/")[2]
+        api_region = configuration_item.get("awsRegion")
+
+        global _client_cache
+        if _client_cache is None:
+            _client_cache = {api_region: get_client(service="apigateway", event=event, region=api_region)}
+
+        api_client = _client_cache.get(api_region)
+
+        # get all resources for this API
+        resources = api_client.get_resources(
+            restApiId=api_id, limit=500,
+            embed=[
+                'methods',
+            ]
+        )
+
+        if "items" not in resources or resources.get("items") is None or len(resources.get("items")) == 0:
+            return 'NOT_APPLICABLE'
+        else:
+            # process each resource to ensure it has an authoriser
+            methods_compliant = True
+            no_auth = 0
+            invalid_auth = 0
+            ok_auth = 0
+
+            for resource in resources.get("items"):
+                if "resourceMethods" in resource:
+                    for method, method_config in resource.get("resourceMethods").items():
+                        resource_method = f"{api_id}:{method}:{resource.get('path')}"
+
+                        auth = method_config.get("authorizationType", None)
+
+                        if auth is None or auth == 'NONE':
+                            methods_compliant = False
+                            no_auth += 1
+                            print(f"No Authorizer found on {resource_method}")
+                        else:
+                            if valid_authorizers is not None:
+                                if auth in valid_authorizers:
+                                    # this authoriser is one of the valid authorisers from the input list
+                                    ok_auth += 1
+                                else:
+                                    methods_compliant = False
+                                    invalid_auth += 1
+                                    print(f"Invalid Authorizer found on {resource_method}")
+                            else:
+                                # no required authoriser types, so we can say this method is OK
+                                ok_auth += 1
+
+            # add the summary evaluation for the stage
+            if methods_compliant:
+                _add_response(stage=configuration_item.get("resourceId"), status="COMPLIANT",
+                              event=event,
+                              comment="All Stage Resource Methods are compliant with Authorizer Policy")
+            else:
+                _add_response(stage=configuration_item.get("resourceId"), status="NON_COMPLIANT",
+                              event=event,
+                              comment=f"Found {no_auth} methods without Authorizers, {invalid_auth} methods without Valid Authorizers, and {ok_auth} with correct Authorizers")
```

Comment:
Those value introduce complexity and API usage which is not required. You can state, "at least 1 method is not using a valid authorizers"